bin/peggy-cli.mjs

```diff
@@ -0,0 +1,593 @@
+import { Command, CommanderError, InvalidArgumentError, Option } from "commander";
+import fs from "fs";
+import path from "path";
+import { default as peggy } from "../lib/peg.js";
+import util from "util";
+
+export { CommanderError, InvalidArgumentError };
+
+// Options that aren't for the API directly:
+const PROG_OPTIONS = ["input", "output", "sourceMap", "test", "testFile", "verbose"];
+const MODULE_FORMATS = ["amd", "bare", "commonjs", "es", "globals", "umd"];
+const MODULE_FORMATS_WITH_DEPS = ["amd", "commonjs", "es", "umd"];
+const MODULE_FORMATS_WITH_GLOBAL = ["globals", "umd"];
+
+// Helpers
+
+function select(obj, sel) {
+  const ret = {};
+  for (const s of sel) {
+    if (Object.prototype.hasOwnProperty.call(obj, s)) {
+      ret[s] = obj[s];
+      delete obj[s];
+    }
+  }
+  return ret;
+}
+
+function commaArg(val, prev) {
+  return (prev || []).concat(val.split(",").map(x => x.trim()));
+}
+
+// Files
+
+function readStream(inputStream) {
+  return new Promise((resolve, reject) => {
+    const input = [];
+    inputStream.on("data", data => { input.push(data); });
+    inputStream.on("end", () => resolve(Buffer.concat(input).toString()));
+    inputStream.on("error", er => {
+      Error.captureStackTrace(er);
```

Comment:
There are a few places where errors coming into async callbacks don't have their stack traces filled in by default, at least on node 17.0.1.  If you specify `--verbose`, the stack trace is printed.  I've put comments on each one.

Reply:
According to the [V8 docs](https://v8.dev/docs/stack-trace-api#stack-trace-collection-for-custom-exceptions) and [this SO answer](https://stackoverflow.com/questions/59625425/understanding-error-capturestacktrace-and-stack-trace-persistance), the stack captured by `Error.captureStackTrace` contains a stack of caller location. So it seems it useless to call it here because we in any case cannot add information about where the `er` is actually was constructed. Besides, I could not find information on whether the existing stack will be rewritten by this call, but if we want to leave it, then this needs to be avoided

Reply:
I thought it gave useful information when I tried it, but I'll just remove these calls.